selfdrive/controls/controlsd.py

```diff
@@ -444,11 +449,22 @@ def state_control(self, CS):
     a_acc_sol = long_plan.aStart + (dt / LON_MPC_STEP) * (long_plan.aTarget - long_plan.aStart)
     v_acc_sol = long_plan.vStart + dt * (a_acc_sol + long_plan.aStart) / 2.0
 
-    # Gas/Brake PID loop
-    actuators.gas, actuators.brake = self.LoC.update(self.active, CS, v_acc_sol, long_plan.vTargetFuture, a_acc_sol, self.CP)
+    if not self.debug_mode:
+      # Gas/Brake PID loop
+      actuators.gas, actuators.brake = self.LoC.update(self.active, CS, v_acc_sol, long_plan.vTargetFuture, a_acc_sol, self.CP)
+
+      # Steering PID loop and lateral MPC
+      actuators.steer, actuators.steeringAngleDeg, lac_log = self.LaC.update(self.active, CS, self.CP, self.VM, params, lat_plan)
+    else:
+      # in joystick debug mode, create dummy lac_log
+      _, _, lac_log = self.LaC.update(self.active, CS, self.CP, self.VM, params, lat_plan)
+      if len(self.sm['testJoystick'].axes) and self.sm['testJoystick'].buttons[1]:  # only when engaged
```

Comment:
Why check the length of the axis? If you want to check if the message was received once you can check if the timestamp is non zero.